core/src/main/scala/cats/ContravariantSemigroupal.scala

```diff
@@ -12,4 +12,7 @@ import simulacrum.typeclass
       def F = self
       def G = Functor[G]
     }
+
+  def contramap2[A, B, C](fb: F[B], fc: F[C])(f: A => (B, C)): F[A] =
```

Comment:
I'm not exactly sure if we need this definition here, as we already have `contramap2`- `contramap22` in the auto-generated `SemigroupalArityFunctions`. WDYT?

Reply:
We could mix in the `SemigroupalArityFunctions` into the companion object of `ContravariantSemigroupal` maybe?

Reply:
Ah, yes! I agree. I didn't realize they were in `SemigroupalArityFunctions`. I'll do that.